WordPress/gutenberg
# Try: Replace BlockDescription component with block type property

Related: #1555

This pull request explores an alternative to #1555, wherein a block can add its description to the inspector by rendering the `<BlockDescription>` component in the block inspector slot. The intent here is to treat description as a base property of a block type, rather than one which is specific to the inspector or edit rendering. The purpose being: a description pairs well with other top-level properties (title, icon), may be useful outside the context of edit, is simpler to document for basic block behaviors, and may need explicit arrangement in the block inspector slot (see #1889 caveats).

There is otherwise no functional difference from the behavior implemented in #1555.

__Testing instructions:__

Verify that the block description is shown in the sidebar "Block" inspector for the selected block.

1. Navigate to Gutenberg > Demo
2. Select a block with a description assigned (e.g. text, image)
3. In the sidebar, switch to Block panel
4. Note that the block description is shown at the top of the block inspector
